flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/YieldingOperatorFactory.java

```diff
@@ -17,10 +17,13 @@
 
 package org.apache.flink.streaming.api.operators;
 
+import org.apache.flink.annotation.PublicEvolving;
+
 /**
  * An operator that needs access to the {@link MailboxExecutor} to yield to downstream operators needs to be created
  * through a factory implementing this interface.
  */
+@PublicEvolving
 public interface YieldingOperatorFactory<OUT> extends StreamOperatorFactory<OUT> {
```

Comment:
This guy should be removed and merged into `StreamOperatorInitializer` by exposing a `Supplier<MailboxExecutor> mailboxExecutorFactory`.

Reply:
That would contradict
```
		// yielding operators cannot be chained to legacy sources
		if (downStreamOperator instanceof YieldingOperatorFactory) {
			// unfortunately the information that vertices have been chained is not preserved at this point
			return !getHeadOperator(upStreamVertex, streamGraph).isStreamSource();
		}
```
What do you think about having mixed pattern? Common parameters in `StreamOperatorParameters`, and "special" or rare ones as decorators?

Reply:
Haven't thought about this case . Since it's `Experimental`, I'm also fine with keeping it for now.
In general, I don't think we should mix patterns though. 
Here is some solution:  `StreamOperatorFactory` could have a `default boolean needsMailboxExecutor() { return false; }`, which triggers a nullable `mailboxExecutor` to be set in `Parameters`. The same method can be used to determine chainability.

Reply:
Maybe let's keep it for now as it is? As I'm not sure if I like the idea of nullifying parameters based on `needsMailboxExecutor()`. I think I would prefer mixed pattern to that.